<?xml version="1.0" ?>
<launch>

  	<!--[x = 2.0 ,y = 5.0] 
  sets the robot in-front of the 1st lane-->
  <arg name="initial_pose_x" default="2.0"/> 
  <arg name="initial_pose_y" default="5.5"/>

  <arg name="initial_pose_a" default="0.0"/>
  <arg name="initial_cov_xx" default="0.0"/>
  <arg name="initial_cov_yy" default="0.0"/>
  <arg name="initial_cov_aa" default="0.0"/>

  <param name="use_sim_time" value="true"/>

  <!--topo-nav arguments for mongoDB-->
  <arg name="topo_map"/> <!--mongodb map name (name ot the map located inside MongoDB with no extension) --> 
  <arg name="db_path"/> <!--/path/to/mongoDB/ -->

  <!--map argument for rasberry_bringup launch file-->
  <arg name="world" value="norwayPolytunnels"/> 

  <!--.yaml map pointing to the pgm map for map_server -->
  <arg name="yaml_map" value="$(find rasberry_gazebo)/maps/norway_poles.yaml"/> 



<!--open mongodb -->
  <include file="$(find mongodb_store)/launch/mongodb_store.launch">
    <arg name="db_path" value="$(arg db_path)"/>
  </include>
  
  <!--rasberry_bringup -->
  <include file="$(find rasberry_bringup)/launch/rasberry_simulation.launch">
   <arg name="world_name" value="$(arg world)"/>
   <arg name="start_pose_x" value="$(arg initial_pose_x)" />
   <arg name="start_pose_y" value="$(arg initial_pose_y)" />
   <arg name="start_pose_z" value="$(arg initial_pose_a)" />
   <arg name="start_pose_R" value="$(arg initial_cov_xx)" />
   <arg name="start_pose_P" value="$(arg initial_cov_yy)" />
   <arg name="start_pose_Y" value="$(arg initial_cov_aa)" />
  </include>


  <!--rasberry_move_base dwa_local_planner-->
  <include file="$(find rasberry_move_base)/launch/move_base_dwa.launch">
   <arg name="initial_pose_x" value="$(arg initial_pose_x)"/>
   <arg name="initial_pose_y" value="$(arg initial_pose_y)"/>
   <arg name="initial_pose_a" value="$(arg initial_pose_a)"/>
   <arg name="initial_cov_xx" value="$(arg initial_cov_xx)"/>
   <arg name="initial_cov_yy" value="$(arg initial_cov_yy)"/>
   <arg name="initial_cov_aa" value="$(arg initial_cov_aa)"/>
   <arg name="map" value="$(arg yaml_map)"/>
  </include>

  <!--topological navigation-->
  <include file="$(find topological_navigation)/launch/topological_navigation.launch">
    <arg name="map" value="$(arg topo_map)"/>
  </include>


  <node pkg="rviz" type="rviz" respawn="false" name="rviz" output="screen" clear_params="true" args="-d $(find rasberry_navigation)/config/norway_topo_nav.rviz">
  </node>


</launch>
